src/components/Icons/Icon.tsx

```diff
@@ -1,105 +1,105 @@
-/* eslint-disable camelcase */
 import * as React from "react";
-import bem from "../../utils/bem";
-import generateUUID from "../../helpers/generateUUID";
+import { Icon as ChakraIcon, Box, useStyleConfig } from "@chakra-ui/react";
 
+import generateUUID from "../../helpers/generateUUID";
 import {
-  IconRotationTypes,
-  IconNames,
-  LogoNames,
+  IconAlign,
   IconColors,
+  IconNames,
+  IconRotationTypes,
   IconSizes,
+  IconTypes,
+  LogoNames,
 } from "./IconTypes";
-
 import iconSvgs from "./IconSvgs";
 
 export interface IconProps {
-  /** Additional attributes to pass to the `<svg>` tag. */
-  attributes?: { [key: string]: any };
-  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
-  blockName?: string;
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
-  modifiers?: string[];
+  /** Aligns the icon. */
+  align?: IconAlign;
   /** className that appears in addition to "icon" */
   className?: string;
-  /** Icons designated as decorative will be ignored by screenreaders */
+  /** Overrides default icon color (black). */
+  color?: IconColors;
+  /** Icons designated as decorative will be ignored by screenreaders. True
+   * by default. */
   decorative?: boolean;
-  /** ID that other components can cross reference for accessibility purposes */
-  id?: string;
-  /** This text will be added as a child `<title>` element inside the `<svg>` tag.  It is recommended to do this for increased accessibility. */
-  titleText?: string;
```

Comment:
There's no way to apply a text title to the icon?  What happens when an icon is more than just decorative?  How is it presented to a screen reader?

Reply:
Forgot to make a note of this:
The `title` attribute in an `svg` element is not accessible. It's useful for other HTML elements but it doesn't work the same for SVGs. The better approach is to add `<title>` HTML elements to the `<svg>` element as its _first_ child. I couldn't find a way to read the SVG component/string and inject a `title` element. I think right now the best approach is to manually update every single SVG element the DS has with a title in their own file. The problem is that in some cases the `titleText` _was_ being passed. Maybe we just make the title static all the time.

To explain the reading issue further, the SVGs we import into `Icon` are all `.svg` files and not React components. This makes it harder to read, decompose, and inject a component or element. I couldn't read the children and put a `<title>` as the first element and push the remaining children after.

We can add a `title` element to each svg file now, but should we agree on a convention? Or for now just put the name of the SVG in the title?

https://css-tricks.com/svg-title-vs-html-title-attribute/

Reply:
Right, we had determined that `title` was not a proper attribute for `svg` and we were able to insert the proper child element into the `svg`.  It took some time, but it was doable.  Did you do make that work or was it Michael?

Icons can be used for so many diffrerent reasons, so I worry about putting a static value in there.  Let's make sure to have a call to talk about this.

Reply:
Hmm I don't think this was ever implemented. The previous iteration added `titleText` to the `title` attribute to the `svg` element.

Reply:
Wow, that's very odd.  I do rememebr this working properly.  Then it must have been Michael.  I recall that he ripped things apart and used a new package to handle the `svg`.  The `svg` came in as a React component and we had much more control over it.  Let me see if I can find it in the history.

Reply:
This was Michael's last commit for this.  There was still an error that was showing in the github tests, but locally it worked fine.

https://github.com/NYPL/nypl-design-system/tree/586ed39c8510f781d65fc7518f0f847c527b12cc/src/components/Icons